src/aggregator/integration/custom_aggregations_test.go

```diff
@@ -79,20 +78,8 @@ func testCustomAggregations(t *testing.T, metadataFns [4]metadataFn) {
 		SetAggregationTypesOptions(aggTypesOpts)
 
 	// Clock setup.
-	var lock sync.RWMutex
-	now := time.Now().Truncate(time.Hour)
-	getNowFn := func() time.Time {
-		lock.RLock()
-		t := now
-		lock.RUnlock()
-		return t
-	}
-	setNowFn := func(t time.Time) {
-		lock.Lock()
-		now = t
-		lock.Unlock()
-	}
-	clockOpts := clock.NewOptions().SetNowFn(getNowFn)
+	testClock := newTestClock(time.Now().Truncate(time.Hour))
+	clockOpts := clock.NewOptions().SetNowFn(testClock.Now)
```

Comment:
Nit: consider creating `testClock.Options()` for this, then you can simply do 
```go
serverOpts = serverOpts.SetClockOptions(testClock.Options()) 
```
everywhere.

Reply:
https://github.com/m3db/m3/pull/3465/commits/3b0741252d23b949648c0bcdb18044317637e82c#diff-ce79fb9967232a7d6b74cc2abd7afa0bf5e3606065f7ad8b959de84f2c5344e8R54-R56